cpp/example_code/cross-service/topics_and_queues/README.md

```diff
@@ -0,0 +1,170 @@
+
+With SQS, messages are received from a queue by polling. With polling, the subscriber receives messages by callling a receive message api. Any code can poll the queue. Also, the messages stay in the queue until you delete them. This gives you more flexibility in how the messages are processed.
+
+The sample code builds a command line application that asks you for input. This is implemented in multiple programming languages, and the interface can vary slightly between languages. The following shows the interface for the C++ implementation.
+
+### Create an SNS topic
+
+```
+Would you like to work with FIFO topics? (y/n) 
+```
+
+You configure FIFO (First-In-First-Out) topics when you create them. Choosing a FIFO topic enables other options, too. To learn more, see [FIFO topics example use case](https://docs.aws.amazon.com/sns/latest/dg/fifo-example-use-case.html).
+
+
+```
+Would you like to use content-based deduplication instead of a deduplication ID? (y/n)
+```
+
+Deduplication is only available for FIFO topics. Deduplication prevents the subscriber from responding more than once to events that are determined to be duplicates. If a message gets published to an SNS FIFO topic and it’s found to be a duplicate within the five-minute deduplication interval, the message is accepted but not delivered. For more information, see [Message deduplication for FIFO topics](https://docs.aws.amazon.com/sns/latest/dg/fifo-message-dedup.html).
+
+Content-based deduplication uses a hash of the content as a deduplication ID. If content-based deduplication is not enabled, you must include a deduplication ID with each message.
+
+```
+Enter a name for your SNS topic:
+```
+
+Topic names can have 1-256 characters. They can contain uppercase and lowercase ASCII letters, numbers, underscores, and hyphens. If you chose a FIFO topic, the application automatically adds a “.fifo” suffix, which is required for FIFO topics.
+
+### Create two SQS queues.
+
+Now, configure two SQS queues to subscribe to your topic.
+
+```
+Enter a name for an SQS queue.
+```
+
+Queue names can have 1-80 characters. They can contain uppercase and lowercase ASCII letters, numbers, underscores, and hyphens. If you chose a FIFO topic, the application automatically adds a “.fifo” suffix, which is required for FIFO queues.
+
+
+```
+Would you like to filter messages for "<queue name>.fifo"s subscription to 
+the topic "<topic name>.fifo"?  (y/n)
+```
+
+If you chose FIFO topics, you can add a filter to the queue’s topic subscription. There are many ways to filter a topic. In this example code, you have the option to filter by a predetermined selection of attributes. For more information about filters, see [Message filtering for FIFO topics](https://docs.aws.amazon.com/sns/latest/dg/fifo-message-filtering.html).
```

Comment:
I notice here it talks about Filters, but it isn't listed in the code comments or the metadata for what the example does. If the user is specifically looking for information on filters, will they be able to find this example?